applications/satellite-imagery-geotiff.ipynb

```diff
@@ -0,0 +1,334 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_scale_factors(filename, band_number):\n",
+    "    with open(filename) as f:\n",
+    "        metadata = json.load(f)\n",
+    "    M_p = metadata['L1_METADATA_FILE'] \\\n",
+    "                  ['RADIOMETRIC_RESCALING'] \\\n",
+    "                  ['REFLECTANCE_MULT_BAND_{}'.format(band_number)]\n",
+    "    A_p = metadata['L1_METADATA_FILE'] \\\n",
+    "                  ['RADIOMETRIC_RESCALING'] \\\n",
+    "                  ['REFLECTANCE_ADD_BAND_{}'.format(band_number)]\n",
+    "    return M_p, A_p"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def calculate_reflectance(ds, band_number, metafile='meta.json'):\n",
+    "    M_p, A_p = load_scale_factors(metafile, band_number)\n",
+    "    toa = M_p * ds + A_p\n",
+    "    return toa"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "red = calculate_reflectance(red, band_number=4)\n",
+    "nir = calculate_reflectance(nir, band_number=5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Because the transformation is composed of arithmetic operations, execution is delayed and the operations are parallelized automatically."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(red.variable.data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The resulting image has floating point data with magnitudes appropriate to reflectance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "red_max, red_min, red_mean = dask.compute(\n",
+    "    red.max(dim=['x', 'y']), \n",
+    "    red.min(dim=['x', 'y']),\n",
+    "    red.mean(dim=['x', 'y'])\n",
+    ")\n",
+    "print(red_max)\n",
+    "print(red_min)\n",
+    "print(red_mean)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Calculate and display NDVI\n",
+    "\n",
+    "Now that we have the image as reflectance values, we are ready to compute NDVI.\n",
+    "\n",
+    "$$\n",
+    "NDVI = \\frac{NIR - Red}{NIR + Red}\n",
+    "$$\n",
+    "\n",
+    "This highlights areas of healthy vegetation with high NDVI values, which appear as green in the image below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ndvi = (nir - red) / (nir + red)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ndvi2d = ndvi.isel(band=0)"
```

Comment:
I think it is clearer to use `squeeze` in this case since the band selection doesn't really mean anything:

```suggestion
    "ndvi2d = ndvi.squeeze()"
```

Reply:
I agree! Thank you for these comments.